Makefile

```diff
@@ -22,18 +22,81 @@ REGISTRY ?= gcr.io/heptio-images
 IMAGE = $(REGISTRY)/$(TARGET)
 DOCKER ?= docker
 DIR := ${CURDIR}
-VERSION ?= v0.1
+VERSION ?= v0.2
+
+ARCH    ?= amd64
+LINUX_ARCHS = amd64 arm64
+PLATFORMS = linux/amd64,linux/arm64
+
+IMAGEARCH ?=
+QEMUARCH  ?=
+
+MANIFEST_TOOL_VERSION := v1.0.0-rc
+
+ifneq ($(ARCH),amd64)
+IMAGE_TAG = $(IMAGE_REPO)-$(ARCH):$(IMAGE_TAG_NAME)
+endif
+
+ifeq ($(ARCH),amd64)
+IMAGEARCH =
+QEMUARCH  = x86_64
+else ifeq ($(ARCH),arm)
+IMAGEARCH = arm32v7/
+QEMUARCH  = arm
+else ifeq ($(ARCH),arm64)
+IMAGEARCH = arm64v8/
+QEMUARCH  = aarch64
+else ifeq ($(ARCH),ppc64le)
+IMAGEARCH = ppc64le/
+QEMUARCH  = ppc64le
+else ifeq ($(ARCH),s390x)
+IMAGEARCH = s390x/
+QEMUARCH  = s390x
+else
+$(error unknown arch "$(ARCH)")
+endif
 
 all: container
 
-container: get_systemd_logs.sh
-	$(DOCKER) build -t $(REGISTRY)/$(TARGET):latest -t $(REGISTRY)/$(TARGET):$(VERSION) .
+pre-cross:
+	docker run --rm --privileged multiarch/qemu-user-static:register --reset
+
+build-container: get_systemd_logs.sh
+	$(DOCKER) build --build-arg IMAGEARCH=$(IMAGEARCH) \
+               --build-arg QEMUARCH=$(QEMUARCH) \
+               -t $(REGISTRY)/$(TARGET):latest -t $(REGISTRY)/$(TARGET):$(VERSION) .
+
+container: pre-cross
+	for arch in $(LINUX_ARCHS); do \
```

Comment:
The logic doesn't seem to follow what I'd expect. For instance, to make the ppc64le image, I'd expect this to work (but it doesnt):
```
ARCH=ppc64le make
```

Instead, it just uses the amd64/arm64 linux_arch values and ignores my `ARCH` value.

I can use:
```
ARCH=ppc64le make build-container
```
but that doesnt tag the image as expected:

```
docker build --build-arg IMAGEARCH=ppc64le/ \
               --build-arg QEMUARCH=ppc64le \
               -t gcr.io/heptio-images/sonobuoy-plugin-systemd-logs:latest -t gcr.io/heptio-images/sonobuoy-plugin-systemd-logs:v0.2 .
```

Maybe we should clarify (in just text) the following behaviors:
 - what gets built/tagged when `make` is run without any env settings in play
 - what gets built/tagged when env vars XYZ are set

It seems like the PR is mostly there but either something is miswired slightly or I'm just not clear on the intended logic/results.

Reply:
Hi @johnSchnake 
The first target in Makefile will be executed when "make" is run, in this case, make = make all.

Reply:
So, if we want to build the specific image for powerpc platform, the command is:
"make build-container ARCH=ppc64le ".
I have fixed the problems of wrong tag.
Please check it.